### PR TITLE
OLH-1389 Redesign the backend for reporting a suspicious activity

### PR DESCRIPTION
**Pull request by andrew-moores**
## Proposed changes

New ADR documenting decision to use Step Functions.

### What changed

New ADR added that documents the decision to use Step Functions.

### Why did it change

The current approach is not as good as that offered by Step Functions.

### Environment variables or secrets

- [x] No environment variables or secrets were added or changed

- [ ] Added parameters to Cloudformation template
- [ ] Added new secret or systems manager parameter
- [ ] Added new environment variable

### Permissions

- [ ] This PR adds or changes permissions

### CODE REVIEW

**Comment by andrew-moores** [comment]
@alex9smith and @andrew-moores to review.

File: docs/adr/0011-reporting-suspicious-activity.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+### Option 3 - single lambda
+
+The whole process could be handled by a single lambda that picks up an event from SQS to decouple the front end.  The lambda would then perform each step required in turn:
+
+1. Update the database.
+2. Create the ticket.
+3. Store the ticket id in the database.
+4. Publish a TxMa event.
+5. Send an email via Notify.
```
Comment (saralk) [inline docs/adr/0011-reporting-suspicious-activity.md:88]: I'm not sure that this order is correct. I would have thought tickets 1, 4 and 5 could all be done in parallel (and have no dependencies on other tickets being done). Steps 2 and 3 are the only ones that have a dependency.

Comment (andrew-moores) [reply docs/adr/0011-reporting-suspicious-activity.md:88]: I made updating the database the first task as we need to ensure a revisit to the activity page provides the correct information about the report.  Agree we could run tasks 1 and 2 in parallel.

File: docs/adr/0011-reporting-suspicious-activity.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+We will allow the user to select one or more records from the activity log to report as suspicious activity.  A suspicious activity report consists of a Zendesk ticket, a TxMA event and an email confirmation.  All of these tasks must finish before a suspicious activity report is considered complete.  Some of the tasks must happen serially and some can be run in parallel.  All of the tasks must be resilient to error. 
+
+The required tasks are:
+
+1. Update the activity log items to show report requested.
+2. Create Zendesk ticket
+3. Update the activity log items with the ticket number
+4. Send an event to TxMA that includes the Zendesk ticket number
+5. Update the activity log items with the TxMA event id
+6. Send an email to Notify.
```
Comment (peterfajemisincabinetoffice) [inline docs/adr/0011-reporting-suspicious-activity.md:16]: As per yesterday's discussion, do we still need the notify steps?

Comment (andrew-moores) [reply docs/adr/0011-reporting-suspicious-activity.md:16]: No we don't but I think having this documented as part of the process still makes a good case for adopting Step Functions.  And the requirements around the emailing are still changing.

Comment (alex9smith) [reply docs/adr/0011-reporting-suspicious-activity.md:16]: Latest update from Ben and TICF is that we're going to switch back to using Notify now, but only send the confirmation email. Let's leave this step in the ADR for now and if we don't end up building it we can consider coming back later and removing it if it's confusing.

**Review by alex9smith — APPROVED** [review]

**Pull request merged**